aya-obj/src/obj.rs

```diff
@@ -947,10 +970,20 @@ impl Object {
                 res?
             }
             BpfSectionKind::Program => {
-                let (program, function) = self.parse_program(&section)?;
-                self.functions.insert(program.function_key(), function);
-                self.programs
-                    .insert(program.section.name().to_owned(), program);
+                let symbols: HashMap<String, Symbol> = self
```

Comment:
We now do this in 3 different places: here, in ::Maps and in ::BtfMaps. We
should do it only once, build `self.symbols_by_section: HashMap<SectionIndex, Vec<usize>>`
while we build `self.symbol_table`. Keys are section indexes, values are symbol
indexes used to index into `self.symbol_table`.

Reply:
Done.